internal/pkg/deploy/cloudformation/cloudformation_integration_test.go

```diff
@@ -258,6 +259,93 @@ func Test_Project_Infrastructure(t *testing.T) {
 			validationFunction(output)
 		}
 	})
+
+	t.Run("Deploys supporting infrastructure for pipeline (KMS Key, S3 Bucket)", func(t *testing.T) {
+		project := archer.Project{Name: randStringBytes(10), AccountID: callerInfo.Account}
+		projectRoleStackName := fmt.Sprintf("%s-infrastructure-roles", project.Name)
+		projectStackSetName := fmt.Sprintf("%s-infrastructure", project.Name)
+
+		// Make sure we delete the stacks after the test is done
+		defer func() {
+			// Clean up any StackInstances we may have created.
+			if stackInstances, err := cfClient.ListStackInstances(&awsCF.ListStackInstancesInput{
+				StackSetName: aws.String(projectStackSetName),
+			}); err == nil && stackInstances.Summaries != nil && stackInstances.Summaries[0] != nil {
+				projectStackInstance := stackInstances.Summaries[0]
+				cfClient.DeleteStackInstances(&awsCF.DeleteStackInstancesInput{
+					Accounts:     []*string{projectStackInstance.Account},
+					Regions:      []*string{projectStackInstance.Region},
+					RetainStacks: aws.Bool(false),
+					StackSetName: projectStackInstance.StackSetId,
+				})
+
+				cfClient.WaitUntilStackDeleteComplete(&awsCF.DescribeStacksInput{
+					StackName: projectStackInstance.StackId,
+				})
+			}
+			// Delete the StackSet once all the StackInstances are cleaned up
+			cfClient.DeleteStackSet(&awsCF.DeleteStackSetInput{
+				StackSetName: aws.String(projectStackSetName),
+			})
+
+			cfClient.DeleteStack(&awsCF.DeleteStackInput{
+				StackName: aws.String(projectRoleStackName),
+			})
+		}()
+
+		// Given our stack doesn't exist
+		roleStackOutput, err := cfClient.DescribeStacks(&awsCF.DescribeStacksInput{
+			StackName: aws.String(projectRoleStackName),
+		})
+
+		require.True(t, len(roleStackOutput.Stacks) == 0, "Stack %s should not exist.", projectRoleStackName)
```

Comment:
If the Stack doesn't exist an error should be returned by `DescribeStacks`. Should we check for an error instead of `len(roleStackOutput.Stacks)`?